src/core/ext/transport/chttp2/client/secure/secure_channel_create.c

```diff
@@ -97,15 +98,15 @@ grpc_channel *grpc_secure_channel_create(grpc_channel_credentials *creds,
   GRPC_API_TRACE(
       "grpc_secure_channel_create(creds=%p, target=%s, args=%p, "
       "reserved=%p)",
-      4, (creds, target, args, reserved));
+      4, ((void *)creds, target, (void *)args, (void *)reserved));
   GPR_ASSERT(reserved == NULL);
   // Make sure security connector does not already exist in args.
```

Comment:
I think that we have 2 solutions here. 

1. Is to do like you suggested but we really have to make sure that the connections to backends have the original target name as the target name of the security connector. That's probably the cleanest solution but it requires one security connector per-subchannel. It's hard for me to judge the performance impact but since premature optimization is the root of all evils, I think this is my favorite solution.

2. Something else we could do is to which is closer to the implementation in this PR is to choose the name to check in the body of the `grpc_security_connector_check_peer` function and call the vtable with the selected name. This would become something like.

In `security_connector.h`:
```c
typedef struct {
  void (*destroy)(grpc_exec_ctx *exec_ctx, grpc_security_connector *sc);

  // Notice the last param which is now 'name_to_check'.
  void (*check_peer)(grpc_exec_ctx *exec_ctx, grpc_security_connector *sc,
                     tsi_peer peer, grpc_auth_context **auth_context,
                     grpc_closure *on_peer_checked, const char *name_to_check);  
} grpc_security_connector_vtable;

// We change the helper function to take in the channel args.
void grpc_security_connector_check_peer(grpc_exec_ctx *exec_ctx,
                                        grpc_security_connector *sc,
                                        tsi_peer peer,
                                        grpc_auth_context **auth_context,
                                        grpc_closure *on_peer_checked,
                                        grpc_channel_args *args);
```

and in the `security_connector.c`, 
```c
void grpc_security_connector_check_peer(grpc_exec_ctx *exec_ctx,
                                        grpc_security_connector *sc,
                                        tsi_peer peer,
                                        grpc_auth_context **auth_context,
                                        grpc_closure *on_peer_checked,
                                        grpc_channel_args *args) {
  if (sc == NULL) {
    grpc_closure_sched(
        exec_ctx, on_peer_checked,
        GRPC_ERROR_CREATE("cannot check peer -- no security connector"));
    tsi_peer_destruct(&peer);
  } else {
    // find the name to check from the channel args whether this is:
    // - the name for the load balancer connection.
    // - the name for a backend/regular connection (original target name).
    // - or NULL if we're on the server side.
    const char *name_to_check = get_name_to_check_from_channel_args(args);
    sc->vtable->check_peer(exec_ctx, sc, peer, auth_context, on_peer_checked,
                           name_to_check);
  }
}
```